gnucash/gnome/dialog-imap-editor.c

```diff
@@ -181,6 +181,40 @@ delete_selected_row (GtkTreeModel *model, GtkTreeIter *iter, ImapDialog *imap_di
         g_free (full_source_account);
 }
 
+static void
+gnc_delete_nomapping (GtkTreeModel *model, GtkTreePath *path,
```

Comment:
this function is defined wrong, look at the definition of GtkTreeMOdelForeachFunc and you see it should be a static gboolean and you will have to return TRUE/FALSE, in your case probably FALSE.